pkgs/ipxe/default.nix

```diff
@@ -25,19 +36,28 @@ in stdenv.mkDerivation rec {
   makeFlags = [
     "ECHO_E_BIN_ECHO=echo" "ECHO_E_BIN_ECHO_E=echo" # No /bin/echo here.
     "EMBED=${embedScript}"
-    "bin/ipxe.kpxe"
-  ];
+  ] ++ lib.optional (legacy) "bin/ipxe.kpxe"
+    ++ lib.optional (efi) "bin-x86_64-efi/ipxe.efi";
 
   preBuild = "cd src";
 
   enableParallelBuilding = true;
 
   installPhase = ''
     mkdir -p $out/nix-support
-    cp bin/ipxe.kpxe $out/ipxe.kpxe
-    chmod 555 $out/ipxe.kpxe
     echo "nix-build out $out" > $out/nix-support/hydra-build-products
-    echo "file binary-dist $out/ipxe.kpxe" >> $out/nix-support/hydra-build-products
+
+    ${lib.optionalString (legacy) ''
+      cp bin/ipxe.kpxe $out/ipxe.kpxe
+      chmod 555 $out/ipxe.kpxe
```

Comment:
copying files and then giving them specific access rights can be done in one `install` command.